servicedispatcher.go

```diff
@@ -67,10 +74,11 @@ func (cs *serviceCommandState) sendMessage(cmd dimse.Message, data []byte) {
 func (disp *serviceDispatcher) findOrCreateCommand(
 	msgID dimse.MessageID,
 	cm *contextManager,
+	activeCommands map[dimse.MessageID]*serviceCommandState,
 	context contextManagerEntry) (*serviceCommandState, bool) {
 	disp.mu.Lock()
 	defer disp.mu.Unlock()
-	if cs, ok := disp.activeCommands[msgID]; ok {
+	if cs, ok := activeCommands[msgID]; ok {
```

Comment:
Pls explain the reason behind this changes.

the `disp.activeCommands` is guarded  by `disp.mu`, so the old code makes sure that every read/write to `disp.activeCommands` must be blocked by `disp.mu.Lock()`